python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -355,9 +356,17 @@ def compute_missing_subset(
         # of the materializability of the asset
         if self.asset_graph.get(asset_key).is_materializable:
             # cheap call which takes advantage of the partition status cache
-            materialized_subset = self._queryer.get_materialized_asset_subset(asset_key=asset_key)
+            partitions_def = self._get_partitions_def(asset_key)
+            if partitions_def:
+                cache_value = AssetStatusCacheValue.blocking_get(self, (asset_key, partitions_def))
+                if cache_value is None:
```

Comment:
a tiny nit, but given that we have so many nested ifs here, it might be easier to read this as:

```
value = cache_value.deserialize_materialized_partition_subsets(partitions_def) if cache_value else partitions_def.empty_subset()
```

The other approach here that I would consider would be to put a method on AssetStatusCacheValue that allows us to just directly get an entity subset.

That way we could do:

```python
cache_value = ...
return cache_value.get_materialized_subset(self) if cache_value else self.get_empty_subset()
```